### PR TITLE
[Metricbeat] Migrate Dropwizard collector Metricset to use ReporterV2Error interface

### PR DESCRIPTION
**Pull request by sayden**
Refer to #10774 for more info

### CODE REVIEW

File: metricbeat/module/dropwizard/collector/_meta/data.json [hunk 1]
```diff
@@ -1,51 +1,34 @@
...
-                "mean": 0,
-                "min": 0,
-                "p50": 0,
-                "p75": 0,
-                "p95": 0,
-                "p98": 0,
-                "p99": 0,
-                "p999": 0,
-                "stddev": 0
+            "_namespace": "testnamespace",
```
Comment (ruflin) [inline metricbeat/module/dropwizard/collector/_meta/data.json]: This should not be here anymore I would assume?

File: metricbeat/module/dropwizard/collector/collector.go [hunk 2]
```diff
@@ -98,17 +97,22 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
...
-		return nil, err
+		return err
 	}
 
 	eventList := eventMapping(dw)
 
 	// Converts hash list to slice
-	events := []common.MapStr{}
 	for _, event := range eventList {
 		event[mb.NamespaceKey] = m.namespace
```
Comment (ruflin) [inline metricbeat/module/dropwizard/collector/collector.go]: This line can probably be removed?

File: metricbeat/module/dropwizard/collector/_meta/data.json (hunk 1 above)
Comment (sayden) [reply metricbeat/module/dropwizard/collector/_meta/data.json]: I thought that we could maintain it just for compatibility but I can remove it, of course.

Comment (ruflin) [reply metricbeat/module/dropwizard/collector/_meta/data.json]: I don't think this was in the previous events.

File: metricbeat/module/dropwizard/collector/collector_integration_test.go [hunk 3]
```diff
@@ -67,19 +71,14 @@ func TestFetch(t *testing.T) {
 	}
 	assert.Equal(t, hasTag, true)
```
Comment (kaiyan-sheng) [inline metricbeat/module/dropwizard/collector/collector_integration_test.go]: Do we only need to check hasTag once after all for _, event := range events {} loops?

File: metricbeat/module/dropwizard/collector/collector_integration_test.go [hunk 4]
```diff
@@ -67,19 +71,14 @@ func TestFetch(t *testing.T) {
...
-	if !assert.NoError(t, err) {
-		t.FailNow()
-	}
 
 	t.Logf("%s/%s event: %+v", f.Module().Name(), f.Name(), events)
 }
 
 func TestData(t *testing.T) {
-	compose.EnsureUp(t, "dropwizard")
+	f := mbtest.NewReportingMetricSetV2Error(t, getConfig())
```
Comment (ruflin) [inline metricbeat/module/dropwizard/collector/collector_integration_test.go]: Now that we have the new data generation, we could remove `TestData`?

File: metricbeat/module/dropwizard/collector/collector_integration_test.go (hunk 3 above)
Comment (sayden) [reply metricbeat/module/dropwizard/collector/collector_integration_test.go]: Uhmmm :thinking: You are right. I'll leave it for a follow up PR, I'm looking forward to finish with the migration and this was like this already anyways :sweat_smile:

File: metricbeat/module/dropwizard/collector/collector_integration_test.go (hunk 4 above)
Comment (sayden) [reply metricbeat/module/dropwizard/collector/collector_integration_test.go]: Absolutely

**Comment by sayden** [comment]
Error is unrelated. Merging

**Pull request merged**